tchannel/tornado/peer.py

```diff
@@ -702,8 +702,18 @@ def choose(self, hostport=None, blacklist=None):
         """
 
         blacklist = blacklist or set()
+        # We we already have the connection in our peer list, use it. Else
+        # don't add it to our peer list.
+        # Refs https://github.com/uber/tchannel-python/issues/464
         if hostport:
-            return self.get(hostport)
+            if hostport in self.hosts:
+                return self.get(hostport)
+            else:
+                return self.peer_class(
+                    tchannel=self.tchannel,
+                    hostport=hostport,
+                    on_conn_change=self._update_heap,
```

Comment:
this peer will get added to the heap when it's updated.  We can set on_conn_change to None and it will get ignored